components/eam/src/physics/crm/crm_physics.F90

```diff
@@ -1184,14 +1224,48 @@ subroutine crm_physics_tend(ztodt, state, tend, ptend, pbuf, cam_in, cam_out, &
 
       call crm_history_out(state, ptend, crm_state, crm_rad, crm_output, crm_ecpp_output, qrs, qrl)
 
+      !---------------------------------------------------------------------------------------------
       ! Convert heating rate to Q*dp to conserve energy across timesteps
+      !---------------------------------------------------------------------------------------------
       do m = 1,crm_nz
          k = pver-m+1
          do i = 1,ncol
             crm_rad%qrad(i,:,:,m) = crm_rad%qrad(i,:,:,m) * state%pdel(i,k) ! for energy conservation
          end do
       end do
 
+      !---------------------------------------------------------------------------------------------
+      ! put rad data back in pbuf
+      !---------------------------------------------------------------------------------------------
+      call pbuf_get_field(pbuf, crm_rad_idx_t,    crm_rad_tmp_temperature)
```

Comment:
Weren't these fields already read from pbuf above? Or was that only if first step?

Reply:
This is another spot that will need to be changed to accommodate pushing the chunk loop down into crm_physics, so I just structured this how it will look when it's inside one of the chunk loops. I'd like to keep it that way even though all these lines will be touched when I change "pbuf" to "pbuf_chunk" and add an indent for the chunk loop.